### ISSUE TITLE
[Accessibility]: Close button in search context dropdown cannot be activated with keyboard

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#34720]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34188

### Problem description

1. Open Sourcegraph
2. Open search context dropdown by focusing on the search context and pressing space or enter
3. Press shift+tab to focus on the close button
4. Press enter or spacebar, observe nothing happens

![image](https://user-images.githubusercontent.com/206864/166071402-6f8eebf7-7b8c-4646-bb85-c2ac6164bc12.png)

### Expected behavior

Dropdown should close

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.
